src/main/java/se/sundsvall/webmessagecollector/service/scheduler/MessageCacheService.java

```diff
@@ -41,12 +59,27 @@ public void fetchMessages(String familyId) {
 		messageRepository.saveAll(messages);
 
 		executionInformationRepository.save(executionInfo);
+		return messages;
 	}
 
-	private ExecutionInformationEntity initiateExecutionInfo(String familyId) {
+	@Transactional
+	public void fetchAttachment(final MessageAttachmentEntity attachmentEntity) {
+		try {
+			final var attachmentStream = openEClient.getAttachment(attachmentEntity.getAttachmentId());
+			if (attachmentStream != null) {
+				attachmentEntity.setFile(new SerialBlob(attachmentStream));
+				messageAttachmentRepository.save(attachmentEntity);
```

Comment:
I thing, if this is called from another transaction scope, that the session is not flushed when the method ends.. If I am not misstaking you should use saveAndFlush.

Reply:
You are correct in this.